.github/workflows/rust.yml

```diff
@@ -16,7 +16,14 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: nightly
     - name: Build
       run: cargo build --verbose
     - name: Run tests
       run: cargo test --verbose
+    - name: Run tests (deterministic based on span)
+      run: RUSTFLAGS='--cfg deterministic_based_on_span' cargo +nightly test --verbose
+    - name: Run tests (deterministic based on span, with sha2)
+      run: RUSTFLAGS='--cfg deterministic_based_on_span' cargo +nightly test --features sha2 --verbose
```

Comment:
Having `deterministic_based_on_span` as a flag but `sha2` as a feature is a bit awkward when dealing with indirect dependencies. 

`deterministic_based_on_span` as a flag makes sense because const_random is likely an indirect dependency of the final application. So its easy to set in one place and have it applied for the entire build. However features need to be declared by the packages that actually use them. 

If an app depends on a library which depends on const-random, and the library does not enable `sha`, (which is likely because it's not on by default) then the app may have a hard time building it the way they want.

Making `sha` a non-optional dependency would work, but it's totally unnecessary if random numbers are being used. 

I'm not sure which is the best way forward.